libauk/src/main/java/com/bitmark/libauk/storage/SecureFileStorage.kt

```diff
@@ -63,8 +70,56 @@ internal class SecureFileStorageImpl(
         return os.toByteArray()
     }
 
-    override fun readOnFilesDir(name: String): ByteArray =
-        read(File(context.filesDir, "$alias-$name").absolutePath)
+    override fun readOnFilesDir(name: String): Single<ByteArray> {
+        var byteArray = byteArrayOf()
+        val isAuthenRequired = BiometricUtil.isAuthenReuired(listOf(name), context)
+        if (isAuthenRequired) {
+            if (context is FragmentActivity) {
+                return BiometricUtil.withAuthenticate<ByteArray>(activity = context,
+                    onAuthenticationSucceeded = { result ->
+                        read(
+                            File(context.filesDir, "$alias-$name").absolutePath,
+                            isAuthenRequired
+                        ).also { byteArray = it }
```

Comment:
assign the byteArray will not have any effect here since it will always return before the value is assigned.